### ISSUE TITLE
Rules Compendium

### ISSUE THREAD

**Issue by D-D-2** [Cards-of-Chaos/Game-Design#1]
So far, we have a Miro board with a collection of ideas for the mechanics each with varying substrates for each (ie. damage types, card abilities). I propose we create a Rules Compendium that we can add to as we develop more of these mechanics.

**Comment by nstevens26** [comment]
I agree, we can make a folder titled "rules" and add the different subcategories of rules. For now I think creating different issues for the type of rule we are working on will help outline the work we have to do.

**Comment by nstevens26** [comment]
Upon reshuffle, players cannot reshuffle spells of level x or higher depending on the characters current level.

**Comment by nstevens26** [comment]
Secondary effects from spells will have a saving throw

**Comment by D-D-2** [comment]
Varying classifications for spell pre-requisites: 

- Some spells are class agnostic 
- Some spells are class specific 
- Some spells gain a benefit if you are a certain subclass.
- No spells require a certain subclass.

**Comment by D-D-2** [comment]
Ability Actions: 

- Add class specific ability action that can act as a "mana-sink" in order to provide players with more options on their turn.

**Comment by D-D-2** [comment]
Spell caps in deck: 

As you level up, the amount of higher level spells in your deck will increase, allowing you to change out the build of your deck without increasing the amount of allotted cards each level. 

Example: 
At level 1, you have 20 first level spells that can go into your deck. 
At level 2, you have 20 first level spells available, and 10 second level spells available. 
At level 3, you have 20 first level spells available, 12 second, and 5 third.

**Comment by D-D-2** [comment]
Deck size: 

The idea now is 20 cards in the deck at the start. This amount stays constant until level 3 or 5. Then the deck size increases to 25 or 30, then at level 7 it increases to 35, then at level 10 it increases to 40. Or something like that.

**Comment by D-D-2** [comment]
Add a passive ability for some classes? For example: Rogue gets a sneak attack passive ability at 1.2-1.3x damage.

**Comment by D-D-2** [comment]
When run out of cards, total mana is reduced by 1

**Comment by D-D-2** [comment]
Long Rest: You can sideboard your entire deck if you want. You regain your full HP. Get your exiled spells back. Regen all ability points.
 
Short Rest: You can regen a certain amount of HP. You can regen a certain amount of cards. You can sideboard a certain amount of cards. Regen certain amount of ability points. Cannot get exiled spells back.
